compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1813,25 +1803,21 @@ fn check_variances_for_type_defn<'tcx>(
     tcx: TyCtxt<'tcx>,
     item: &hir::Item<'tcx>,
     hir_generics: &hir::Generics<'tcx>,
-) {
+) -> Result<(), ErrorGuaranteed> {
     let identity_args = ty::GenericArgs::identity_for_item(tcx, item.owner_id);
 
     match item.kind {
         ItemKind::Enum(..) | ItemKind::Struct(..) | ItemKind::Union(..) => {
             for field in tcx.adt_def(item.owner_id).all_fields() {
-                if field.ty(tcx, identity_args).references_error() {
-                    return;
-                }
+                field.ty(tcx, identity_args).error_reported()?;
```

Comment:
I wonder if we should just use `error_reported` to delay the error message below *but* don't early-return (using `emit_unless`). I kinda feel bad that we're setting *all* of the substs to bivariant even if they are constrained otherwise (e.g. to covariance) somewhere else in the definition.

Reply:
Yea, these early aborts were preexisting. I didn't want to touch them in the previously-big PR (I'm assuming they got added to avoid ICEs), but now it's reasonable to investigate.